### PR TITLE
Use display name instead of user ID when rendering power events (PSC-82)

### PR DESCRIPTION
**Pull request by Johennes**
Notes: Use display name instead of user ID when rendering power events 

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Use display name instead of user ID when rendering power events ([\#9295](https://github.com/matrix-org/matrix-react-sdk/pull/9295)).

### CODE REVIEW

File: src/TextForEvent.tsx
Comment (t3chguy) [inline src/TextForEvent.tsx:469]: This won't handle disambiguation properly, if there are 3 people in the room called `Alice` it'll say something like

> Changed the power level of Alice to 51, Alice to 90, Alice to 91.

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/TextForEvent.tsx
Comment (Johennes) [reply src/TextForEvent.tsx:469]: Good catch! I missed that in the other places in this file it's used as e.g. `ev.sender?.name || getRoomMemberDisplayname(ev)`. 🤦‍♂️

How about we change the implementation of `getRoomMemberDisplayname` to first try the `name` property?

```
diff --git a/src/TextForEvent.tsx b/src/TextForEvent.tsx
index b94f7361e0..6be8dee332 100644
--- a/src/TextForEvent.tsx
+++ b/src/TextForEvent.tsx
@@ -54,7 +54,7 @@ function getRoomMemberDisplayname(event: MatrixEvent, userId = event.getSender()
     const client = MatrixClientPeg.get();
     const roomId = event.getRoomId();
     const member = client.getRoom(roomId)?.getMember(userId);
-    return member?.rawDisplayName || userId || _t("Someone");
+    return member?.name || member?.rawDisplayName || userId || _t("Someone");
 }
```

Comment (t3chguy) [reply src/TextForEvent.tsx:469]: Maybe, it needs careful looking into as some places handle disambiguation themselves (e.g. the timeline where it is rendered in a different font size & colour)

Comment (Johennes) [reply src/TextForEvent.tsx:469]: Looks like the only other place this function is used is

```
function textForMemberEvent(ev: MatrixEvent, allowJSX: boolean, showHiddenEvents?: boolean): () => string | null {
    // XXX: SYJS-16 "sender is sometimes null for join messages"
    const senderName = ev.sender?.name || getRoomMemberDisplayname(ev);
    const targetName = ev.target?.name || getRoomMemberDisplayname(ev, ev.getStateKey());
```

in the same file. So I think this change is safe due to the preceding `name ||`.

**Review by t3chguy — APPROVED** [review]
Looks sane to me

**Comment by t3chguy** [comment]
> Here's what your changelog entry will look like:

> ✨ Features

> + Use display name instead of user ID when rendering power events (PSC-82) (https://github.com/matrix-org/matrix-react-sdk/pull/9295).

Can we please not add the bracketed identifiers to our changelogs?

**Comment by Johennes** [comment]
> > Here's what your changelog entry will look like:
> 
> > ✨ Features
> 
> > * Use display name instead of user ID when rendering power events (PSC-82) ([Use display name instead of user ID when rendering power events (PSC-82) #9295](https://github.com/matrix-org/matrix-react-sdk/pull/9295)).
> 
> Can we please not add the bracketed identifiers to our changelogs? They make no sense to users

@t3chguy yes, sorry, this slipped. Have fixed the notes now.

**Comment by Johennes** [comment]
Hm, `should click top left of view source event toggle ` failed but I'm not sure how that would relate to my changes.

**Comment by t3chguy** [comment]
@Johennes https://github.com/vector-im/element-web/issues/23270

**Comment by Johennes** [comment]
@t3chguy did you want me to do anything else before this can be merged?

**Comment by t3chguy** [comment]
The tests need to pass to unlock the merge button, unfortunately a case of restarting and hoping until someone fixes the flake.

**Comment by Johennes** [comment]
All green now 🙂

**Pull request merged**
